plugins/pi4j-plugin-linuxfs/src/main/java/com/pi4j/plugin/linuxfs/provider/spi/LinuxFsSpi.java

```diff
@@ -364,33 +357,62 @@ public int write(byte b) {
     public int write(byte[] data, int offset, int length) {
         Objects.checkFromIndexSize(offset, length, data.length);
 
+        int fullEntries = (length < SPI_BUFFSIZ) ? 1 : length / SPI_BUFFSIZ;
+        int partialEntries = (((fullEntries * SPI_BUFFSIZ) < length) ? 1 : 0);
+        int numberXferEntries = fullEntries + partialEntries;
```

Comment:
This can be written without the two helper variables as

int numberXferEntries = (length + SPI_BUFSIZ -1) / SPI_BUFSIZ;